source/extensions/omni.isaac.lab/omni/isaac/lab/envs/direct_rl_env.py

```diff
@@ -298,11 +298,12 @@ def step(self, action: torch.Tensor) -> VecEnvStepReturn:
             self._apply_action()
             # set actions into simulator
             self.scene.write_data_to_sim()
-            render = self._sim_step_counter % self.cfg.sim.render_interval == 0 and (
-                self.sim.has_gui() or self.sim.has_rtx_sensors()
-            )
             # simulate
-            self.sim.step(render=render)
+            self.sim.step(render=False)
+            if self._sim_step_counter % self.cfg.sim.render_interval == 0 and (
+                self.sim.has_gui() or self.sim.has_rtx_sensors()
+            ):
+                self.sim.render()
```

Comment:
```suggestion
            self.sim.step(render=False)
                        # render between steps only if the GUI or an RTX sensor needs it
            # note: we assume the render interval to be the shortest accepted rendering interval.
            #    If a camera needs rendering at a faster frequency, this will lead to errors
            if self._sim_step_counter % self.cfg.sim.render_interval == 0 and (
                self.sim.has_gui() or self.sim.has_rtx_sensors()
            ):
                self.sim.render()
```